### ISSUE TITLE
segfault when running hash_join.sql test script

### ISSUE THREAD

**Issue by pbhowmic** [cmu-db/peloton#290]
I get a segmentation fault when running the hash_join.sql test script. Looks like peloton is unable to handle a join on an empty table. Script run thru psql

> postgres=# \i script/testing/join/hash_join.sql
> DROP TABLE IF EXISTS A 0
> DROP TABLE IF EXISTS B 0
> SET ENABLE_MERGEJOIN TO FALSE 0
> SET ENABLE_NESTLOOP TO FALSE 0
> SET ENABLE_HASHJOIN TO TRUE 0
> CREATE TABLE A(id INT, value INT) 0
> CREATE TABLE B(id INT, value INT) 0
> psql:script/testing/join/hash_join.sql:13: server closed the connection unexpectedly
>   This probably means the server terminated abnormally
>   before or while processing the request.
> psql:script/testing/join/hash_join.sql:13: connection to server was lost

The offending line in the script (line 13) is the last one in the script below

```
-- create the test tables
DROP TABLE IF EXISTS A;
DROP TABLE IF EXISTS B;

SET ENABLE_MERGEJOIN TO FALSE;
SET ENABLE_NESTLOOP TO FALSE;
SET ENABLE_HASHJOIN TO TRUE;

CREATE TABLE A(id INT, value INT);
CREATE TABLE B(id INT, value INT);

-- join with empty tables
SELECT * FROM A INNER JOIN B ON A.value = B.value;
```

**Comment by eric-haibin-lin** [comment]
Hmmm. Unfortunately INNER JOIN queries always fail since we don't have a optimizer for join queries yet. It fails also when the table is non-empty. Same for other kinds of joins. 
You can see how optimizer handles select stmt now: 
https://github.com/cmu-db/peloton/blob/master/src/optimizer/simple_optimizer.cpp#L91

Actually the hash_join.sql should be revised as well, some of the queries are not what peloton expects since we changed our parser. https://github.com/cmu-db/peloton/blob/master/src/parser/sql_parser.y#L892

For example, `SELECT * FROM A FULL OUTER JOIN B ON A.value = B.value ` should be modified as:  `SELECT * FROM A OUTER JOIN B ON A.value = B.value`. 

Is this a blocker to what you're working on?

**Comment by apavlo** [comment]
This will be fixed by #243

**Comment by pbhowmic** [comment]
@eric-haibin-lin Not killing me right now, just trying to get a feel for the code before I dive in, but I don;t want to misclassify any errors I run into, I just need to know when the error is due to a pre-existing bug and not because of any code I change or add.

**Comment by poojanilangekar** [comment]
Fixed by #243
